src/fragments/lib/graphqlapi/js/working-with-files.mdx

```diff
@@ -0,0 +1,984 @@
+amplify add auth             #Select default configuration
+amplify add storage          #Select S3 with read/write access
+amplify add api              #Select Cognito User Pool for authorization type
+```
+
+When prompted, use the following schema:
+
+```graphql
+type Song @model {
+  id: ID!
+  name: String!
+  covertArtKey: String
+}
+```
+
+Save the schema and run `amplify push` to deploy the changes.
+
+To use files and attachments you need authorization credentials for reading and writing to Amazon S3, which `amplify add auth` configures in the default setting along with a Cognito User Pool. For more details on Storage / API authorization, see the [Configuring Authorization](#configuring-authorization) section below.
+
+For the complete working example, including required imports and React component state management, see the [Complete Example](#complete-example) below.
+
+## Creating a record with a GraphQL API
+
+Running the following mutation will create a record with the GraphQL API. The corresponding file will be associated with this record.
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function createSong() {
+  const songDetails: CreateSongInput = {
+    name: `My first song`,
+  };
+
+  try {
+    const response = await API.graphql<GraphQLQuery<CreateSongMutation>>({
+      query: mutations.createSong,
+      variables: { input: songDetails },
+    });
+    setCurrentSong(response?.data?.createSong);
+  } catch (error) {
+    console.error("Error creating song: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function createSong() {
+  const songDetails = {
+    name: `My first song`,
+  };
+
+  try {
+    const response = await API.graphql({
+      query: mutations.createSong,
+      variables: { input: songDetails },
+    });
+    setCurrentSong(response?.data?.createSong);
+  } catch (error) {
+    console.error("Error creating song: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Query the record with the GraphQL API
+
+The following helper function will be used to retrieve the record from the GraphQL API in the examples below:
+
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+async function getSong() {
+  if (!currentSong) return;
+
+  try {
+    const oneSong = await API.graphql<GraphQLQuery<GetSongQuery>>({
+      query: queries.getSong,
+      variables: { id: currentSong.id },
+    });
+
+    return oneSong.data?.getSong;
+  } catch (error) {
+    console.error("Error retrieving song: ", error);
+  }
+}
+``` 
+
+</Block>
+<Block name="JavaScript">
+
+```js
+async function getSong() {
+  if (!currentSong) return;
+
+  try {
+    const oneSong = await API.graphql({
+      query: queries.getSong,
+      variables: { id: currentSong.id },
+    });
+
+    return oneSong.data?.getSong;
+  } catch (error) {
+    console.error("Error retrieving song: ", error);
+  }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+## Uploading the file to S3
```

Comment:
Feedback attached above. Let's combine the snippets between Storage and API workflows and not separate it out. One recurring theme from customer conversations and that came up in the Discord office hours is "What's the sequence of events I should do for Storage + API, which one goes first? Is there guidance?"

Reply:
Just want to make sure we answer that question "head-on"